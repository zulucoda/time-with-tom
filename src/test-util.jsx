import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
