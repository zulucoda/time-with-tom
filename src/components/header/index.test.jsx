import React from 'react';
import { render } from '@testing-library/react';
import Header from '.';
import { ReactRouterWrapperTestUtil } from '../../test-util';

describe('Components - Header - Unit Test', function () {
  test('does it render', () => {
    const { container } = render(
      <ReactRouterWrapperTestUtil>
        <Header />
      </ReactRouterWrapperTestUtil>,
    );
    expect(container).toMatchSnapshot();
  });
});
