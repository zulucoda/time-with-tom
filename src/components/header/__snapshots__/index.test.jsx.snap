// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Header - Unit Test does it render 1`] = `
<div>
  <div
    class="sc-AxjAm kmVibe"
  >
    <h1
      class="sc-AxirZ dxeCUY"
    >
      Time with Tom
    </h1>
    <ul
      class="sc-AxiKw TgnyB"
    >
      <li
        class="sc-AxhCb hLIlzG"
      >
        <a
          href="/schedule"
        >
          <span
            class="sc-AxhUy kLxUxn"
          >
            schedule
          </span>
        </a>
      </li>
      <li
        class="sc-AxhCb hLIlzG"
      >
        <a
          href="/booking-request"
        >
          <span
            class="sc-AxhUy kLxUxn"
          >
            booking request
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
